lib/puppet/pops/types/type_factory.rb

```diff
@@ -181,13 +181,7 @@ def self.any
   # @api public
   #
   def self.regexp(pattern = nil)
-    if pattern
-      t = PRegexpType.new(pattern.is_a?(Regexp) ? pattern.inspect[1..-2] : pattern)
-      t.regexp unless pattern.nil? # compile pattern to catch errors
-      t
-    else
-      PRegexpType::DEFAULT
-    end
+    pattern ?  PRegexpType.new(pattern) : PRegexpType::DEFAULT
```

Comment:
Can you elaborate - I don't quite understand. Do you mean that it accepts `//opt` as input in string form, but when presenting that adds option wrappers?